### PR TITLE
add timeouts to prevent slowloris attacks

### PR DESCRIPTION
**Pull request by sanikachavan5**
This pull request introduces security enhancements and configurability improvements for HTTP server timeouts in the agent and connect proxy, aiming to mitigate Slowloris and related denial-of-service attacks. It adds new configuration options for HTTP timeouts, ensures they are correctly parsed and validated, and updates both the agent and proxy to use these values. Comprehensive tests are added to verify correct behavior and edge cases.
### Description

**Security and HTTP Timeout Enhancements:**

* Added configuration options for HTTP server timeouts (`read_timeout`, `read_header_timeout`, `write_timeout`, `idle_timeout`) to `HTTPConfig` and `RuntimeConfig`, with sensible defaults and minimums, to protect against Slowloris and similar attacks (`agent/config/config.go`, `agent/config/runtime.go`, `agent/config/builder.go`, `agent/agent.go`). [[1]](diffhunk://#diff-1308de62937b63bdfe5da80b0bf5216c4916d12ad7a40aee4843649c1ca9c02fR698-R701) [[2]](diffhunk://#diff-4a299b76809f423d4204a02ca5965a2bedd8973939134daee19201311d26bbf0R800-R831) [[3]](diffhunk://#diff-77452bdff6ee872bdd7b08a2603dfb39d165d45c5bcafc5f8bccaaf010aa7e85R936-R939) [[4]](diffhunk://#diff-62b6ad581fe3a3059ae8c85ef0f31dde4092bfdecfa7d6857c470bcacaa8cc8bR1171-R1174)
* Updated the agent's HTTP server initialization to use these configurable timeout values (`agent/agent.go`).
* Updated the connect proxy's pprof HTTP server to use a configurable timeout (`--pprof-timeout`) and set minimums for security (`command/connect/proxy/proxy.go`). [[1]](diffhunk://#diff-a7cb8ba3e736b091d99d316f521725e7c84d6c3184e90bf924de8da88c318de3R59) [[2]](diffhunk://#diff-a7cb8ba3e736b091d99d316f521725e7c84d6c3184e90bf924de8da88c318de3R93-R96) [[3]](diffhunk://#diff-a7cb8ba3e736b091d99d316f521725e7c84d6c3184e90bf924de8da88c318de3L156-R179)

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

## PCI review checklist

- [ ] I have documented a clear reason for, and description of, the change I am making.

- [ ] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.

- [ ] If applicable, I've documented the impact of any changes to security controls.

  Examples of changes to security controls include using new access control methods, adding or removing logging pipelines, etc.

### CODE REVIEW

File: website/content/docs/reference/agent/configuration-file/index.mdx [hunk 1]
```diff
@@ -87,6 +87,25 @@ Examples:
 
 Refer to the [formatting specification](https://golang.org/pkg/time/#ParseDuration) for additional information.
 
+#### Duration validation
```
Comment (boruszak) [inline website/content/docs/reference/agent/configuration-file/index.mdx]:
```suggestion
#### Duration formatting validation
```

File: website/content/docs/reference/agent/configuration-file/index.mdx [hunk 2]
```diff
@@ -87,6 +87,25 @@ Examples:
...
+
+When configuring timeout and duration values, Consul validates the format and enforces minimum values where applicable. Invalid duration strings will cause the agent to fail to start with a descriptive error message.
```
Comment (boruszak) [inline website/content/docs/reference/agent/configuration-file/index.mdx]:
```suggestion
When you configure timeout and duration values, Consul validates the formatting and enforces minimum values where applicable. When you use invalid duration strings, the agent will fail to start and the output will include a descriptive error message.
```

File: website/content/docs/reference/agent/configuration-file/index.mdx [hunk 3]
```diff
@@ -87,6 +87,25 @@ Examples:
...
+
+**Valid examples:**
+- `"10s"` - 10 seconds
+- `"1.5m"` - 1.5 minutes (90 seconds)
+- `"2h30m45s"` - Complex duration with hours, minutes, and seconds
+- `"0s"` - Zero duration (where allowed)
```
Comment (boruszak) [inline website/content/docs/reference/agent/configuration-file/index.mdx]:
```suggestion
**Examples of valid duration formatting**:

- `"10s"` - 10 seconds
- `"1.5m"` - 1.5 minutes, equivalent to 90 seconds (`90s`)
- `"2h30m45s"` - Complex duration with hours, minutes, and seconds
- `"0s"` - Zero duration, although not all fields support this value.
```

File: website/content/docs/reference/agent/configuration-file/index.mdx [hunk 4]
```diff
@@ -87,6 +87,25 @@ Examples:
...
+
+**Invalid examples that will cause errors:**
+- `"10"` - Missing unit specification
+- `"10x"` - Invalid unit (`x` is not a recognized unit)
+- `""` - Empty string
+- `"abc"` - Non-numeric string
+- `"-5s"` - Negative durations (not accepted for timeout configurations)
```
Comment (boruszak) [inline website/content/docs/reference/agent/configuration-file/index.mdx]:
```suggestion
**Examples of invalid formatting that produce errors**:

- `"10"` - Missing unit specification
- `"10x"` - Invalid unit, as `x` is not a recognized duration
- `""` - Empty string
- `"abc"` - Non-numeric string
- `"-5s"` - Negative durations are not accepted for timeout configurations
```

File: website/content/docs/reference/agent/configuration-file/index.mdx [hunk 5]
```diff
@@ -87,6 +87,25 @@ Examples:
...
+
+**Minimum value enforcement:** Some timeout configurations enforce minimum values to prevent misconfigurations that could cause connection issues or security vulnerabilities. For example, handshake timeouts typically require a minimum of 1 second. When a configured value is below the minimum, Consul will report a validation error during startup.
```
Comment (boruszak) [inline website/content/docs/reference/agent/configuration-file/index.mdx]:
```suggestion
Some timeout configurations enforce minimum values to prevent connection issues or security vulnerabilities. For example, handshake timeouts typically require a minimum of 1 second. When a configured value is below the minimum, Consul reports a validation error during startup.
```

**Review by boruszak — DISMISSED** [review]
I reviewed the docs updates and left suggestions to align with style guidelines. Otherwise LGTM!

**Comment by sanikachavan5** [comment]
@boruszak I have made the required changes, can you take a look?
@dduzgun-security please take a look, conflicts are resolved.

**Review by santoshpulluri — COMMENTED** [review]
Please update the description.

**Review by boruszak — DISMISSED** [review]
Docs LGTM! 

Approved on behalf of consul-docs

File: agent/http_test.go [hunk 6]
```diff
@@ -113,7 +113,11 @@ func TestHTTPServer_UnixSocket_FileExists(t *testing.T) {
 		t.SkipNow()
 	}
 
-	tempDir := testutil.TempDir(t, "consul")
+	tempDir, err := os.MkdirTemp("", "consul_sock_test_")
```
Comment (dduzgun-security) [inline agent/http_test.go:116]: Any specific reason why we moved away from the `testutil.TempDir` function?

File: command/connect/proxy/proxy.go [hunk 7]
```diff
@@ -153,8 +160,23 @@ func (c *cmd) Run(args []string) int {
...
 		go func() {
 			c.UI.Output(fmt.Sprintf("Starting pprof HTTP endpoints on "+
-				"http://%s/debug/pprof", c.pprofAddr))
-			log.Fatal(http.ListenAndServe(c.pprofAddr, nil))
+				"http://%s/debug/pprof (timeout: %v)", c.pprofAddr, c.pprofTimeout))
+
+			// Use minimum timeout values for security while respecting user configuration
+			readTimeout := c.pprofTimeout
+			if readTimeout < 1*time.Second {
+				readTimeout = 1 * time.Second
```
Comment (dduzgun-security) [inline command/connect/proxy/proxy.go:168]: Is a default of 1 second enough for the pprof timeout?

File: agent/http_test.go [hunk 8]
```diff
@@ -1818,7 +1822,11 @@ func TestHTTPServer_HandshakeTimeout(t *testing.T) {
 		buf := make([]byte, 10)
 		_, err = conn.Read(buf)
 		require.Error(r, err)
-		require.Contains(r, err.Error(), "EOF")
+		// After adding timeouts to prevent slowloris attacks, the connection
+		// times out with "i/o timeout" instead of "EOF"
+		if !strings.Contains(err.Error(), "EOF") && !strings.Contains(err.Error(), "i/o timeout") {
```
Comment (dduzgun-security) [inline agent/http_test.go]: Not sure about this condition. Can we check to see the returned value from `err.Error()` and adjust the tests accordingly to use `require.Contains` like before?

**Review by dduzgun-security — CHANGES_REQUESTED** [review]

File: agent/http_test.go (hunk 6 above)
Comment (sanikachavan5) [reply agent/http_test.go:116]: testutil.TempDir creates paths like: /tmp/TestHTTPServer_UnixSocket_FileExists-consul2397590942/test.sock
This path was 113 characters - exceeding the Unix socket limit
Result: bind: invalid argument error when creating the socket

File: command/connect/proxy/proxy.go (hunk 7 above)
Comment (sanikachavan5) [reply command/connect/proxy/proxy.go:168]: User Default: pprofTimeout defaults to 30 seconds (line 90)
Security Floor: If user sets < 1 second, it's bumped up to 1 second (line 168)

**Review by dduzgun-security — DISMISSED** [review]

File: agent/config/runtime.go [hunk 9]
```diff
@@ -797,6 +797,38 @@ type RuntimeConfig struct {
...
+	// hcl: http_config { read_header_timeout = "10s" }
+	HTTPReadHeaderTimeout time.Duration
+
+	// HTTPWriteTimeout is the maximum duration before timing out writes of the response.
+	// This timeout prevents slow response drain attacks.
+	// A zero or negative value means there will be no timeout.
+	//
+	// Default: 30s, Minimum: 1s
+	// hcl: http_config { write_timeout = "30s" }
+	HTTPWriteTimeout time.Duration
```
Comment (anandmukul93) [inline agent/config/runtime.go:831]: does this setting affect pprof collection for durations greater than 30 sec. is pprof fired and run in background with a response immediately provided ? then it should be fine. Just check and update the testing once.

**Review by anandmukul93 — DISMISSED** [review]
LGTM , 
Test the pprof behavior for write back. i think it acks immediately . just update the test for a longer duration testing scenario. like 2 min, 30 sec interval.

File: agent/agent_test.go [hunk 10]
```diff
@@ -6469,6 +6469,49 @@ func assertDeepEqual(t *testing.T, x, y interface{}, opts ...cmp.Option) {
 	}
 }
 
+func TestAgent_HTTPServerTimeouts(t *testing.T) {
+	if testing.Short() {
```
Comment (anandmukul93) [inline agent/agent_test.go]: why testing short required in these changes ?

File: command/connect/proxy/proxy.go [hunk 11]
```diff
@@ -153,8 +160,23 @@ func (c *cmd) Run(args []string) int {
 	if c.pprofAddr != "" {
 		go func() {
 			c.UI.Output(fmt.Sprintf("Starting pprof HTTP endpoints on "+
-				"http://%s/debug/pprof", c.pprofAddr))
-			log.Fatal(http.ListenAndServe(c.pprofAddr, nil))
+				"http://%s/debug/pprof (timeout: %v)", c.pprofAddr, c.pprofTimeout))
+
+			// Use minimum timeout values for security while respecting user configuration
+			readTimeout := c.pprofTimeout
+			if readTimeout < 1*time.Second {
```
Comment (anandmukul93) [inline command/connect/proxy/proxy.go:167]: readTimeout := cmp.Max(c.pprofTimeout, time.Second)

File: command/connect/proxy/proxy.go [hunk 12]
```diff
@@ -153,8 +160,23 @@ func (c *cmd) Run(args []string) int {
 	if c.pprofAddr != "" {
 		go func() {
 			c.UI.Output(fmt.Sprintf("Starting pprof HTTP endpoints on "+
-				"http://%s/debug/pprof", c.pprofAddr))
-			log.Fatal(http.ListenAndServe(c.pprofAddr, nil))
+				"http://%s/debug/pprof (timeout: %v)", c.pprofAddr, c.pprofTimeout))
```
Comment (anandmukul93) [inline command/connect/proxy/proxy.go:163]: https://github.com/hashicorp/consul-enterprise/blob/ed92a4608531e26bea7dc101a4706cd3774d1e93/command/connect/proxy/proxy.go#L210-L214

this whole use case is for a go-proxy that we fire in test cases as far as i understand. it actually doesnt run an envoy proxy but rather no proxy but a handler with default mux inside /net/http/. lets be sure if we really need this change for now as this is only for testing right ?

File: .changelog/22739.txt [hunk 13]
```diff
@@ -0,0 +1,3 @@
+```release-note:security
+security: Configure HTTP server timeouts to prevent Slowloris denial-of-service attacks on agent HTTP endpoints and connect proxy pprof endpoints
```
Comment (anandmukul93) [inline .changelog/22739.txt]: connect proxy is for testing server. shouldnt be required here in release changes.

**Comment by anandmukul93** [comment]
IMPORTANT - my questtion regarding the timeout from command to the pprof handler for an agent still remains whether that would be affected by the 30 sec timeouts ? 
Code pointer here - 
[agent handler](https://github.com/hashicorp/consul-enterprise/blob/44ab494d494515128ddd5d74fd1a035d3ac187c9/agent/agent.go#L1172-L1177)

File: command/connect/proxy/proxy.go (hunk 11 above)
Comment (sanikachavan5) [reply command/connect/proxy/proxy.go:167]: cmp has no function Max

File: command/connect/proxy/proxy.go (hunk 12 above)
Comment (sanikachavan5) [reply command/connect/proxy/proxy.go:163]: this one internally will use the same pprof endpoints, so this is fine i guess

**Review by dduzgun-security — APPROVED** [review]
LGTM, good that these values would be configurable in case someone wants to change them. 

It would be good to ping the #team-nomad Slack just to make them aware of this change in case they need to edit something in their pipelines or deployments prior to merging this change.

**Review by anandmukul93 — APPROVED** [review]
lgtm

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20,1.21] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.18,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sanikachavan5! a backport is missing for this PR [22739] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```
